test/selenium_tests/framework.py

```diff
@@ -110,7 +110,11 @@ def write_file(name, content, raw=False):
                 buf.write(content.encode("utf-8") if not raw else content)
 
         os.makedirs(target_directory)
-        self.driver.save_screenshot(os.path.join(target_directory, "last.png"))
+        try:
+            self.driver.save_screenshot(os.path.join(target_directory, "last.png"))
+        except Exception:
+            print("Failed to dump screenshot for error debugging.")
+            write_file("screenshot_exception.txt", traceback.format_exc())
         write_file("page_source.txt", self.driver.page_source)
         write_file("DOM.txt", self.driver.execute_script("return document.documentElement.outerHTML"))
         write_file("stacktrace.txt", traceback.format_exc())
```

Comment:
This should be moved before the try/except or will print the traceback from the except block instead of the original one.

Reply:
Is that how that works? I thought maybe because we have left the inner exception by this point. What you are saying though makes good sense - I'll try to reorganize this a bit. Thanks for the feedback.

Reply:
Unfortunately yes, that's how it works:
```
In [15]: try:
   ....:     raise Exception("foo")
   ....: except Exception:
   ....:     try:
   ....:         raise Exception("bar")
   ....:     except Exception:
   ....:         print traceback.format_exc()
   ....:     print traceback.format_exc()
   ....:     
Traceback (most recent call last):
  File "<ipython-input-15-06117ef92270>", line 5, in <module>
    raise Exception("bar")
Exception: bar

Traceback (most recent call last):
  File "<ipython-input-15-06117ef92270>", line 5, in <module>
    raise Exception("bar")
Exception: bar
```

Reply:
Good example - thanks @nsoranzo!